### ISSUE TITLE
Provide lots of hooks via hook library

### ISSUE THREAD

**Issue by lvjr** [lvjr/tabularray#197]
We need to provide lots of hooks via `hook` library:
```latex
tabularray/begin
tabularray/end
tabularray/head/begin
tabularray/head/end
tabularray/foot/begin
tabularray/foot/end
tabularray/body/begin
tabularray/body/end
tabularray/body/trial/begin
tabularray/body/trial/end
tabularray/body/final/begin
tabularray/body/final/end
tabularray/row/begin
tabularray/row/end
tabularray/row/head/begin
tabularray/row/head/end
tabularray/row/body/begin
tabularray/row/body/end
tabularray/row/foot/begin
tabularray/row/foot/end
tabularray/cell/begin
tabularray/cell/end
tabularray/cell/trial/begin
tabularray/cell/trial/end
tabularray/cell/final/begin
tabularray/cell/final/end
```
See #27, #29, #146, #159, #179.

**Comment by marvlg** [comment]
Good idea! Is there any reason to put them in a library? Why not provide them by default?

**Comment by lvjr** [comment]
Hooks make the package run slower.  And hooks can not work in TeX Live 2020 on Overleaf.

**Comment by michal-h21** [comment]
For the HTML conversion with TeX4ht, I need to be able to detect features like colspan, rowspan, cell alignment, colors and borders. It seems that most of these information (except for borders, probably?) are available in the `\__tblr_build_cell_content:NN` macro, where I can use them for inline CSS for each table cell. 

The one problem that I have found so far is how to detect a cell that shouldn't be displayed.  In HTML, when you use 
`<td collspan="2">`, you shouldn't include next `<td>`, because it would be spurious. The same is true for `rowspan`, the `<td>` that is bellow the one that uses `rowspan`, shouldn't be used. So we need to detect somehow that the current cell is hidden. It shouldn't be hard to do that on the TeX4ht side, but I wonder if it already isn't available in `tabularray` and I just missed it. It is also possible that tagged PDF will need such functionality as well. 

For example:

```
\documentclass{article}
\usepackage{tabularray}
\begin{document}

\begin{tblr}{|c|c|c|c|c|}
\hline
\SetCell[r=2]{c} 2 Rows
& \SetCell[c=2]{c} 2 Columns
& & \SetCell[r=2,c=2]{c} 2 Rows 2 Columns & \\
\hline
& 2-2 & 2-3 & & \\
\hline
3-1 & 3-2 & 3-3 & 3-4 & 3-5 \\
\hline
\end{tblr}
\end{document}
```

**Comment by lvjr** [comment]
I need to solve issue #179 first before starting to handle this issue. So hooks will not arrive soon.

**Comment by lvjr** [comment]
In the above commit, I have added libraries `hook` and `html` into `trial/tabularray.sty`. All public hooks and variables are documented in
https://github.com/lvjr/tabularray/wiki/HooksAndVariables

**Comment by lvjr** [comment]
In issue #547, all key paths start with `tblr`, but all hook paths here start with `tabularray`, maybe be we should unify them. But what should we use, `tblr` or `tabularray`?

**Comment by lvjr** [comment]
> In issue #547, all key paths start with `tblr`, but all hook paths here start with `tabularray`, maybe we should unify them. But what should we use, `tblr` or `tabularray`?

Since some users have already used these hooks, it is better not to change their names. (I have changed the planned key paths in #547.) But we can create two new commands `\AddToTblrHook` and `\AddToTblrHookNext`.

**Comment by lvjr** [comment]
We can add two new variables: `\lTblrRowHeadBool` and `\lTblrRowFootBool`.

**Comment by lvjr** [comment]
I have added public varaibles `\lTblrRowHeadInt` (from `rowhead`) and `\lTblrRowFootInt` (from `rowfoot`).

**Comment by lvjr** [comment]
No, you can't. And since users have to input the `rowhead` value, you may define a new environment to get the value and use it.
